jscomp/core/lam_compile_external_call.ml

```diff
@@ -106,8 +106,8 @@ let ocaml_to_js_eff ~(arg_label : Melange_ffi.External_arg_spec.label_noname)
   in
   match arg_type with
```

Comment:
this function shows why we can't optimize nested `[@mel.uncurry]` implicitly. Here's the explanation:

- whenever we apply arguments to an `external` function, we walk both the `external` argument list (as declared by the user), and the params passed to the function
- nested `[@mel.uncurry]` are, however, and by definition, not parameters that we pass, but rather, arguments to some callback. Take the simplest example:

```ocaml
external foo : (((unit -> unit)[@mel.uncurry]) -> unit) -> unit = "foo"

let () = foo (fun f -> f ())
```

You can see how, in the function application, `f` is passed to us by the callback (the one we actually pass as a parameter). We don't have that information later on to apply `f` in the uncurried way.